src/core/src/pass/low_latency.cpp

```diff
@@ -93,13 +93,13 @@ std::shared_ptr<ov::opset9::Assign> replace_with_memory(const ov::Input<ov::Node
     using namespace ov::opset9;
     using namespace ov::op::util;
 
-    VariableInfo var_info{ov::PartialShape::dynamic(), ov::element::dynamic, variable_name};
-    auto variable = std::make_shared<Variable>(var_info);
-
     ov::Output<ov::Node> read_value_in = input.get_source_output();
     if (use_const_initializer) {
         read_value_in = create_init_subgraph(read_value_in, to);
     }
+
+    VariableInfo var_info{read_value_in.get_partial_shape(), read_value_in.get_element_type(), variable_name};
+    auto variable = std::make_shared<Variable>(var_info);
```

Comment:
```suggestion
    auto variable = std::make_shared<Variable>(read_value_in.get_partial_shape(), read_value_in.get_element_type(), variable_name);
```
the `std::make_shared` should use ctor and make dynamic object in-place.

Reply:
double-checked, it doesn't work